edge-modules/DirectMethodSender/DirectMethodSenderBase.cs

```diff
@@ -49,12 +49,12 @@ public async Task<HttpStatusCode> InvokeDirectMethodAsync(CancellationTokenSourc
 
                 logger.LogInformation($"Invoke DirectMethod with count {this.directMethodCount}: finished.");
                 this.directMethodCount++;
-                return (HttpStatusCode)resultStatus;
+                return new Tuple<HttpStatusCode, long>((HttpStatusCode)resultStatus, this.directMethodCount);
```

Comment:
should it be better to make this.directMethodCount as protected, so derived class can access it instead of return from method?

Reply:
Correct me if I misunderstanding but currently the derived classes `DirectMethodCloudSender` and `DirectMethodLocalSender` is `this.directMethodCount` agnostic. The reason this function has to return the value as a tuple because the program (caller) of the method needs a way to access the parameter for reporting.

Reply:
Your understanding is correct, my question is why don't make direct method count accessible by derived classes if all derived classes need to use it?